docs/source/documentation/r_scripts.rst

```diff
@@ -0,0 +1,117 @@
+(relative) file paths to data.
+
+The outputs of the scripts are written to stdout, and will be parsed later by the Python script.
+
+Below is an example of what your file should look like:
+
+.. code-block:: R
+    :caption: my_script.R
+
+    #!/usr/bin/env Rscript
+    args <- commandArgs()
+    # your script here
+    ...
+    write(outputs, "")
+
+
+Calling the R script from Python
+--------------------------------
+The Python script passed to Substra wraps the R script, so that it can be executed as a Python subprocess.
+The Python script reads the inputs defined as Substra ``FunctionInputSpec``, converts everything to string,
+appends all parameters in a command (``subprocess.run`` expects a list of str) and launched the subprocess.
+After the subprocess has finished, the output is cleaned.
+Everything printed to stdout in the R script is in the string ``raw_output.stdout``, depending on the type of output,
+additional cleaning steps might be required.
+Finally, the output is saved as a pickle file, to be shared with other nodes in the organisation.
+
+.. code-block:: Python
+    :caption: python_wrapper.py
+
+    import pickle
+    import subprocess
+    import substratools as tools
+
+
+    @tools.register
+    def run_script(inputs, outputs, task_properties):
+        data_file = inputs["data_file_path"]
+        param1 = str(inputs["param1"])
+        param2 = str(inputs["param2"])
+        raw_output = subprocess.run(['Rscript', 'my_script.R', data_file, param1, param2], capture_output=True)
+        model = int(raw_output.stdout.strip())
+        save_model(model, outputs["model"])
+
+
+    def save_model(model, path):
+        with open(path, "wb") as f:
+            pickle.dump(model, f)
+
+
+    if __name__ == "__main__":
+        tools.execute()
+
+Adapting the opener
+-------------------
+When using Substra with Python, the ``Opener`` object is used to load the data in memory.
+When using R, we don't need to load the data as Python objects in memory, so the opener simply returns the file path (or paths).
+
+.. code-block:: Python
+    :caption: opener.py
+
+    import pathlib
+    import substratools as tools
+
+    import os
+
+    class StubOpener(tools.Opener):
+        def fake_data(self, n_samples=None):
+            return ""
+
+        def get_data(self, folders):
+            return list(pathlib.Path(folders[0]).glob("*.csv"))
+
+
+Writing the Dockerfile
+----------------------
+We modify the Dockerfile to install R in the container, and copy both R and Python scripts.
+
+
+.. code-block:: Dockerfile
+    :caption: Dockerfile
+
+    # this base image works in both CPU and GPU enabled environments
+    FROM ghcr.io/substra/substra-tools:0.20.0-nvidiacuda11.8.0-base-ubuntu22.04-python3.9
+
+    # install R
+    RUN apt-get update \
+     && apt-get -y install r-base
+
+    # add your algorithm scripts to docker image
+    ADD my_script.R .
+    ADD python_wrapper.py .
```

Comment:
Do we expect them to change often the `python_wrapper.py`? I expect them to change it at least less often than the R script, therefore I would move it one line up.

Reply:
good catch 👍